# db_constraints.db_exceptions

[](Generated by Ddoc from source/db_constraints/db_exceptions.d)

The db_exceptions module contains:
  + [UniqueConstraintException](#UniqueConstraintException)
  + [KeyedException](#KeyedException)
  + [CheckConstraintException](#CheckConstraintException)
  + [ForeignKeyException](#ForeignKeyException)

**License:**
[GPL-2.0](https://github.com/marmy28/db_constraints/blob/master/LICENSE)


**Authors:**
Matthew Armbruster


**Source:** [source/db_constraints/db_exceptions.d](https://github.com/marmy28/db_constraints/tree/master/source/db_constraints/db_exceptions.d)



***

<a id="UniqueConstraintException"></a>
```d
class UniqueConstraintException: object.Exception;

```

Inherits from Exception. This is thrown whenever
there is a unique constraint violation.

***

<a id="UniqueConstraintException.this"></a>
```d
this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null);

```
Parameters |
---|
*string msg*|
&nbsp;&nbsp;&nbsp;&nbsp;the message thrown with the unique constraint violation|
*string file*|
&nbsp;&nbsp;&nbsp;&nbsp;the file where the exception occurred|
*size_t line*|
&nbsp;&nbsp;&nbsp;&nbsp;the line number where the exception occurred|
*Throwable next*|
&nbsp;&nbsp;&nbsp;&nbsp;references the exception that was being handled when this one was generated|





***

<a id="KeyedException"></a>
```d
class KeyedException: object.Exception;

```

Inherits from Exception. This is thrown whenever
there is an exception dealing with the keyed items.

***

<a id="KeyedException.this"></a>
```d
this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null);

```
Parameters |
---|
*string msg*|
&nbsp;&nbsp;&nbsp;&nbsp;the message thrown|
*string file*|
&nbsp;&nbsp;&nbsp;&nbsp;the file where the exception occurred|
*size_t line*|
&nbsp;&nbsp;&nbsp;&nbsp;the line number where the exception occurred|
*Throwable next*|
&nbsp;&nbsp;&nbsp;&nbsp;references the exception that was being handled when this one was generated|





***

<a id="CheckConstraintException"></a>
```d
class CheckConstraintException: object.Exception;

```

Inherits from Exception. This is thrown whenever
there is a check constraint violation.

***

<a id="CheckConstraintException.this"></a>
```d
this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null);

```
Parameters |
---|
*string msg*|
&nbsp;&nbsp;&nbsp;&nbsp;the message thrown with the check constraint violation|
*string file*|
&nbsp;&nbsp;&nbsp;&nbsp;the file where the exception occurred|
*size_t line*|
&nbsp;&nbsp;&nbsp;&nbsp;the line number where the exception occurred|
*Throwable next*|
&nbsp;&nbsp;&nbsp;&nbsp;references the exception that was being handled when this one was generated|





***

<a id="ForeignKeyException"></a>
```d
class ForeignKeyException: object.Exception;

```

Inherits from Exception. This is thrown whenever
there is an exception dealing with the foreign keys.

***

<a id="ForeignKeyException.this"></a>
```d
this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null);

```
Parameters |
---|
*string msg*|
&nbsp;&nbsp;&nbsp;&nbsp;the message thrown with the foreign key|
*string file*|
&nbsp;&nbsp;&nbsp;&nbsp;the file where the exception occurred|
*size_t line*|
&nbsp;&nbsp;&nbsp;&nbsp;the line number where the exception occurred|
*Throwable next*|
&nbsp;&nbsp;&nbsp;&nbsp;references the exception that was being handled when this one was generated|







Copyright :copyright: 2015 | Page generated by [Ddoc](http://dlang.org/ddoc.html) on Sat Oct 10 10:55:04 2015

